### PR TITLE
Fix constexpr complaints on test vectors

### PR DESCRIPTION
**Pull request by cecille**
Fixes a compiler error on local where const values are being used to initialize a constexpr bytespan.

### CODE REVIEW

**Review by ksperling-apple — APPROVED** [review]

File: src/credentials/BUILD.gn [hunk 1]
```diff
@@ -126,6 +128,23 @@ static_library("credentials") {
...
 
+source_set("test_paa_store") {
+  sources = [
+    "attestation_verifier/CHIPAttCert_test_paa_store.cpp",
+    "attestation_verifier/CHIPAttCert_test_paa_store.h",
+  ]
+
+  public_deps = [
+    "${chip_root}/src/lib/core:chip_config_header",  # for lib/support/Span.h
+    "${nlassert_root}:nlassert",  # for lib/support/Span.h
```
Comment (bzbarsky-apple) [inline src/credentials/BUILD.gn:139]: Followup: Span.h should actually be in a source_set that has the right deps, so consumers don't have to do this.

File: src/credentials/attestation_verifier/CHIPAttCert_test_paa_store.h
Comment (bzbarsky-apple) [inline src/credentials/attestation_verifier/CHIPAttCert_test_paa_store.h:1]: I think it's fine to just call these TestPAAStore.h and TestPAAStore.cpp, but perhaps document at the top of the h file what it's storing....

**Review by bzbarsky-apple — APPROVED** [review]

File: src/credentials/BUILD.gn (hunk 1 above)
Comment (cecille) [reply src/credentials/BUILD.gn:139]: darwin is the problem here - I think I just force updated, but I did have it as its own source set, and there's a circular dependency between span and the darwin logging that I cannot figure out how to untangle.

Comment (cecille) [reply src/credentials/BUILD.gn:139]: Especially since I have to push it through the CI to compile darwin.

Comment (bzbarsky-apple) [reply src/credentials/BUILD.gn:139]: Right, that's why I said "Followup"!  Please just file the issue.  If you have a diff that works except for Darwin, please point me to it and I will sort out the Darwin bits.

File: src/credentials/attestation_verifier/CHIPAttCert_test_paa_store.h
Comment (cecille) [reply src/credentials/attestation_verifier/CHIPAttCert_test_paa_store.h:1]: Changed.

File: src/credentials/BUILD.gn (hunk 1 above)
Comment (bzbarsky-apple) [reply src/credentials/BUILD.gn:139]: @cecille PR up at https://github.com/project-chip/connectedhomeip/pull/29491

**Review by tehampson — APPROVED** [review]

**Pull request merged**